internal/app/cf-terraforming/cmd/util.go

```diff
@@ -168,200 +169,89 @@ func flattenAttrMap(l []interface{}) map[string]interface{} {
+						if len(stepChild.Body().Attributes()) != 0 || len(stepChild.Body().Blocks()) != 0 {
+							parent.AppendBlock(stepChild)
+						}
+					}
+				case map[string]interface{}:
+					processBlocks(schemaBlock.NestedBlocks[block].Block, s, child.Body(), block)
+				case []interface{}:
+					for _, nestedItem := range s {
+						stepChild := hclwrite.NewBlock(block, []string{})
+						processBlocks(schemaBlock.NestedBlocks[block].Block, nestedItem.(map[string]interface{}), stepChild.Body(), block)
+						if len(stepChild.Body().Attributes()) != 0 || len(stepChild.Body().Blocks()) != 0 {
+							parent.AppendBlock(stepChild)
+						}
+					}
+				default:
+					log.Debugf("unable to generate recursively nested blocks for %T", s)
 				}
-			case []map[string]interface{}:
-				for i, attr := range attribute.([]map[string]interface{}) {
-					indexedQuery := fmt.Sprintf("%v.%v", query, i)
-					nestedBlockOutput := nestBlocks(schemaBlock.NestedBlocks[block].Block, attributes, indexedQuery)
-					repeatedBlockOutput := writeNestedBlock(sortedInnerAttributes, schemaBlock.NestedBlocks[block].Block, attr)
-					appendBlock(&output, block, nestedBlockOutput, repeatedBlockOutput)
+				if len(child.Body().Attributes()) != 0 || len(child.Body().Blocks()) != 0 {
+					parent.AppendBlock(child)
 				}
-			case map[string]interface{}:
-				nestedBlockOutput := nestBlocks(schemaBlock.NestedBlocks[block].Block, attributes, query)
-				repeatedBlockOutput := writeNestedBlock(sortedInnerAttributes, schemaBlock.NestedBlocks[block].Block, attribute.(map[string]interface{}))
-				appendBlock(&output, block, nestedBlockOutput, repeatedBlockOutput)
-			default:
-				log.Debugf("unexpected attribute struct type %T for block %s", attribute, block)
 			}
 		} else {
-			log.Debugf("nested mode %q for %s not recognised", schemaBlock.NestedBlocks[block].NestingMode, block)
-		}
-	}
-
-	return output
-}
-
-func appendBlock(output *string, block, nestedBlockOutput, repeatedBlockOutput string) {
-	if repeatedBlockOutput != "" {
-		*output += block + " {\n"
-		*output += repeatedBlockOutput
-
-		if nestedBlockOutput != "" {
-			*output += nestedBlockOutput
-		}
-		*output += "}\n"
-	} else if repeatedBlockOutput == "" && nestedBlockOutput != "" {
-		*output += block + " {\n"
-
-		if nestedBlockOutput != "" {
-			*output += nestedBlockOutput
-		}
-		*output += "}\n"
-	}
-}
-
-func writeNestedBlock(attributes []string, schemaBlock *tfjson.SchemaBlock, attrStruct map[string]interface{}) string {
-	nestedBlockOutput := ""
-
-	for _, attrName := range attributes {
-		ty := schemaBlock.Attributes[attrName].AttributeType
-
-		switch {
-		case ty.IsPrimitiveType():
-			switch ty {
-			case cty.String, cty.Bool, cty.Number:
-				nestedBlockOutput += writeAttrLine(attrName, attrStruct[attrName], false)
-			default:
-				log.Debugf("unexpected primitive type %q", ty.FriendlyName())
+			if parentBlock == "" && block == "id" {
```

Comment:
gotta be careful here as nested `id`s are ok but top level are reserved keywords.

Reply:
this should only skip if its the root body, and we see an ID. other ids should generate just fine.